email-service/internal/notifications/user_notifier.go

```diff
@@ -0,0 +1,115 @@
+package notifications
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+
+	"github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/email-service/internal/models"
+	"github.com/VictoriaMetrics/metrics"
+)
+
+var logger *slog.Logger
+
+func getLogger() *slog.Logger {
+	if logger == nil {
+		logger = slog.Default().With(slog.String("src", "userNotifier"))
+	}
+	return logger
+}
+
+var (
+	notificationsSentTotal   = metrics.NewCounter(`notifications_sent_total`)
+	notificationsFailedTotal = metrics.NewCounter(`notifications_failed_total`)
+)
+
+type EmailClient interface {
+	SendEmail(ctx context.Context, emails []string, subject, message string) error
+}
+
+type SubscriberRepository interface {
```

Comment:
Interfaces with one method should _typically_ end with `er` suffix.

`os.Reader` `os.Writer` `fmt.Stringer`.

`Repository` is a big conception, but you need only one method from it. Why don't you call it `SubscriberFinder` ?